src/DBFTables.jl

```diff
@@ -308,4 +353,116 @@ function Base.getproperty(dbf::Table, name::Symbol)
     return colarr
 end
 
+
+Base.write(io::IO, dbf::Table) = Base.write(io, getfield(dbf, :header), getfield(dbf, :data), 0x1a)
+Base.write(path::AbstractString, dbf::Table) = open(io -> Base.write(io, dbf), touch(path), "w")
+
+
+"Generic .dbf writer for the Tables.jl interface."
+write(path::AbstractString, tbl) = open(io -> write(io, tbl), touch(path), "w")
+
+function write(io::IO, tbl)
+    fields, records = get_field_descriptors(tbl)
+    fieldcolumns = Dict{Symbol,Int}(f.name => i for (i,f) in enumerate(fields))
+    hsize = UInt16(length(fields) * 32 + 32)
+    rsize = UInt16(sum(x -> x.length, fields)) + 1
+
+    version = 0x03
+    last_update = today()
+    incomplete = false
+    encrypted = false
+    mdx = false
+    lang_id = 0x00
+
+    h = Header(version, last_update, records, hsize, rsize, incomplete, encrypted, mdx, lang_id, fields, fieldcolumns)
+    out = Base.write(io, h)
+
+    for row in Tables.rows(tbl)
+        out += write_record(io, fields, row)
+    end
+    out += Base.write(io, 0x1a)  # EOF marker
+    return out
+end
+
+function get_field_descriptors(tbl)
+    fields = FieldDescriptor[]
+    dct = Tables.dictcolumntable(tbl)
+    sch = Tables.schema(dct)
+    for (name, type) in zip(sch.names, sch.types)
+        ndec = 0x0
+        len = 0x0
+        dbf_type = 'C'
+        T = Base.nonmissingtype(type)
+        if T isa Date
+            dbf_type = 'D'
+            len = 0x08
+        elseif T <: AbstractString
+            # TODO: support memos.  Currently strings > 254 bytes will error
+            len = UInt8(maximum(x -> length(string(x)), dct[name]))
+            if len > 254
+                @warn "Strings will be truncated to 254 characters."
+                len = 254
+            end
+            dbf_type = 'C'
+        elseif type === Float64
+            dbf_type = 'O'
+            len = 0x08
+            ndec = 0x01
+        elseif T <: AbstractFloat
+            dbf_type = 'F'
+            len = UInt8(20)
+            ndec = 0x01
+        elseif T <: Bool
+            dbf_type = 'L'
+            len = 0x1
+        elseif T <: Date
+            dbf_type = 'D'
+            len = 0x8
+        elseif T <: Integer
+            dbf_type = 'N'
+            len = UInt8(maximum(x -> length(string(x)), dct[name]))
+        elseif T <: AbstractFloat
```

Comment:
Seems like we won't hit this codepath since we already have an `elseif T <: AbstractFloat` above?

Reply:
Good catch!  I'll remove it.